### ISSUE TITLE
unexpected behaviour when getting QIX service with 2 different configs

### ISSUE THREAD

**Issue by pouc** [qlik-oss/enigma.js#17]
### Description

When getting the qix service with two different config objects, for instance different mixins, the second enigma instance gets mixins from the first instance. Disabling caching solves the issue.

### Steps to Reproduce

```javascript
var config = {
    schema: qixSchema,
    host: 'https://localhost',
    session: {
        route: 'app/engineData',
        host: 'localhost',
        port: 4747,
        unsecure: false
    },
    mixins: [{
        types: ['Doc'],
        init: (args) => {},
        extend: {
            test1: function() {
                console.log('test1');
                return 1;
            }
    }],
    createSocket(url) {
        [...]
    },
};

enigma.getService('qix', config).then((qix) => {
    return qix.global.getDocList().then((docList) => {
        return promise.all(
            docList.map((doc) => {
                return qix.global.openDoc(doc.qDocId).then(app => {
                    return app.test1();
                });
            })
        );
    });
});

// Prints test1 for each available doc on the platform

config = {
    schema: qixSchema,
    host: 'https://localhost',
    session: {
        route: 'app/engineData',
        host: 'localhost',
        port: 4747,
        unsecure: false
    },
    mixins: [{
        types: ['Doc'],
        init: (args) => {},
        extend: {
            test2: function() {
                console.log('test2');
                return 2;
            }
    }],
    createSocket(url) {
        [...]
    },
};

return enigma.getService('qix', config).then((qix) => {
    return qix.global.getDocList().then((docList) => {
        return promise.all(
            docList.map((doc) => {
                return qix.global.openDoc(doc.qDocId).then(app => {
                    return app.test2();
                });
            })
        );
    });
});

// Fails because test2 is not a function
// In fact test2 is not defined on "app" because the openDoc function returned
// a cached session from first step
```

##### Expected behavior

I would expect cached sockets to be private to enigma instances (meaning not shared)

##### Actual behavior

Cache is shared

##### Library

* [X] Node.js
* [ ] Browser

##### Operating system

* [X] Windows
* [ ] OSX
* [ ] Linux

##### Qlik Sense

* [ ] Desktop
* [X] Enterprise

**Comment by peol** [comment]
Hi! Thanks for the report :)

By default enigma.js tries to optimize instances/sessions by returning an existing one (based on the config), if it exist. You can opt out of this behavior by setting `config.session.disableCache` to `true`: https://github.com/qlik-oss/enigma.js/blob/master/docs/qix/configuration.md#config-object

Let us know if you still have trouble after settings that configuration option.

**Comment by pouc** [comment]
Hi Andrée, thanks for the reply.

I have noticed that disabling caching solved the issue.

I raised the issue because I had the feeling that caching should not be enabled between 2 different instances of the qix service.

Especially when these don't share the same config!

Best

**Comment by peol** [comment]
You make a good point, we'll add it to our list :)

**Comment by marcusoffesson** [comment]
In a future version 2 of enigma, we would like to remove session caching altogether. See #66.

**Comment by peol** [comment]
As stated, this will be fixed in v2.0, among a bunch of other improvements. See the referenced issue for more information.
